src/main/java/com/bmc/truesight/saas/meter/client/model/Event.java

```diff
@@ -56,6 +86,84 @@ public static Event of(String title, Severity severity, String message) {
                 .build();
     }
 
+    /**
+     * Creates an Event with the given required parameters: title, severity, message and properties.
+     *
+     * @param title      the Title of this event (must not be greater than 255 characters long).
+     * @param severity   the Severity of the event (info, warn, error or critical)
+     * @param message    a message for this event (must not be greater than 255 characters long).
+     * @param properties a Map of properties Key and Values, must not be null
+     * @return a new Immutable Event built with the title, severity, message and properties provided.
+     */
+    public static Event of(String title, Severity severity, String message, Map<String, String> properties) {
+        return ImmutableEvent.builder()
+                .title(title)
+                .severity(severity)
+                .message(message)
+                .properties(properties)
+                .build();
+    }
+
+    /**
+     * Creates an Event with the given title, severity, message, properties, senderRef, sourceRef, timestamp, tags, at and ad.
+     *
+     * @param title      the Title of this event (must not be greater than 255 characters long).  Required.
+     * @param severity   the Severity of the event (info, warn, error or critical). Defaults to info if not specified.
+     * @param message    a message for this event, must not be greater than 255 characters long (if not null).
+     * @param properties a Map of properties Key and Values. Can be null.
+     * @param sourceRef  the source of this event, can be null. If null defaults to the Fully Qualified Domain Name of the meter host.
+     * @param senderRef  the sender of this event, can be null. If null the meter determines this value.
+     * @param timestamp  time of when this Event was created, can be null. If null defaults to <a href="https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#now--">Instant.now()</a>.
+     * @param tags       a Set of String values - specifically an com.google.common.collect.ImmutableSet &lt;String&gt;.  Can be null.
+     * @param at         App Data Type - Opaque field that is application specific.  Can be used to provide type for app data. Can be null.
+     * @param ad         App Data - Application specific data.  In most cases this should be base64 encoded. Can be null.
+     * @return a new Immutable Event built with the title, severity, message, properties, senderRef, sourceRef, timestamp, tags, at and ad provided.
+     */
+    public static Event of(String title, Severity severity, String message, Map<String, String> properties, String sourceRef, String senderRef, Instant timestamp, ImmutableSet<String> tags, String at, String ad) {
+        ImmutableEvent.Builder immutableEventBuilder = ImmutableEvent.builder().title(title);
```

Comment:
this method is unnecessary. the immutables library provides us with builders more complicated events.